### PR TITLE
Fix/config file

### PR DESCRIPTION
**Pull request by munna0908**
# Description

This PR facilitates PSDK users to start a node using a configuration file, Currently we support only `json` based configuration file. 

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [x] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

File: command/helper/config.go [hunk 1]
```diff
@@ -14,26 +14,24 @@ import (
...
-	BlockGasTarget string                        `json:"block_gas_target"`
-	GRPCAddr       string                        `json:"rpc_addr"`
-	JSONRPCAddr    string                        `json:"jsonrpc_addr"`
-	Telemetry      *Telemetry                    `json:"telemetry"`
-	Network        *Network                      `json:"network"`
-	SecretsManager *secrets.SecretsManagerConfig `json:"secrets_manager"`
-	Seal           bool                          `json:"seal"`
-	TxPool         *TxPool                       `json:"tx_pool"`
-	LogLevel       string                        `json:"log_level"`
-	Consensus      map[string]interface{}        `json:"consensus"`
```
Comment (zivkovicmilos) [inline command/helper/config.go:33]: What happened to this consensus configuration?

We are using it for some pending PRs, namely the PoA / PoS implementation

File: command/helper/config.go [hunk 2]
```diff
@@ -94,7 +90,15 @@ func (c *Config) BuildConfig() (*server.Config, error) {
...
 	conf.DataDir = c.DataDir
+	// Set the secrets manager config if it was passed in
+	if c.Secrets != "" {
+		secretsConfig, readErr := secrets.ReadConfig(c.Secrets)
+		if readErr != nil {
+			return nil, fmt.Errorf("unable to read config file, %v", readErr)
+		}
 
+		conf.SecretsManager = secretsConfig
+	}
```
Comment (zivkovicmilos) [inline command/helper/config.go:104]: You read the config here, but it was never saved to this configuration struct.

Please notice you're only saving it into a temporary var here, but that's about it:
https://github.com/0xPolygon/polygon-sdk/blob/f4d9896f0b2dc70bb3226f0149c6fbb0676dcd3a/command/helper/helper.go#L432

**Review by zivkovicmilos — CHANGES_REQUESTED** [review]
There have been a few oversights that render some functionality useless now.

Please address the comments.

Otherwise it looks good 👍

File: command/helper/config.go (hunk 2 above)
Comment (munna0908) [reply command/helper/config.go:104]: Thanks for the catch , fixed it [here](https://github.com/0xPolygon/polygon-sdk/pull/270/commits/0c7ba6270159d89fcaa6a3f9bf6e3e8548ed6481)

File: command/helper/config.go (hunk 1 above)
Comment (munna0908) [reply command/helper/config.go:33]: I couldn't find any use in current implementation, Restored it [here](https://github.com/0xPolygon/polygon-sdk/pull/270/commits/562271a4e44449e9c3dfe1fd1c31df0401ea8407)

**Review by Kourin1996 — APPROVED** [review]

**Review by zivkovicmilos — APPROVED** [review]
Thank you for clearing some things up 🙏 

Lookin' good 💯

**Review by lazartravica — APPROVED** [review]

**Comment by lazartravica** [comment]
@munna0908 Nice job, feel free to merge this and the docs PR.

**Pull request merged**